### PR TITLE
move nodeData length check after latestTally is updated in CalculateAtRestData

### PR DESCRIPTION
**Pull request by cam-a**
What: move nodeData length check after latestTally is updated

Why: In the event that we only have inline segments, SaveBucketTallies fails consistently because if there are no remote segments we return out of CalculateAtRestData before the timestamp is updated. The timestamp never changes, and we get a failure on unique constraint when trying to enter the bucket tallies into the database

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: pkg/accounting/tally/tally_test.go [hunk 1]
```diff
@@ -59,6 +59,58 @@ func TestDeleteRawBefore(t *testing.T) {
...
+
+		// Setup: The data in this tally should match the pointer that the uplink.upload created
+		expectedTally := accounting.BucketTally{
+			Segments:       1,
+			InlineSegments: 1,
+			Files:          1,
+			InlineFiles:    1,
+			Bytes:          expectedTotalBytes,
+			InlineBytes:    expectedTotalBytes,
+			MetadataSize:   111, // brittle, this is hardcoded since its too difficult to get this value progamatically
```
Comment (jenlij) [inline pkg/accounting/tally/tally_test.go]: why is this one 111 rather than 112?

File: pkg/accounting/tally/tally_test.go [hunk 2]
```diff
@@ -59,6 +59,58 @@ func TestDeleteRawBefore(t *testing.T) {
...
+		assert.NoError(t, err)
+
+		// Run calculate twice to test unique constraint issue
+		for i := 0; i < 2; i++ {
+			latestTally, actualNodeData, actualBucketData, err := tallySvc.CalculateAtRestData(ctx)
+			require.NoError(t, err)
+			assert.Len(t, actualNodeData, 0)
+
+			_, err = planet.Satellites[0].DB.Accounting().SaveBucketTallies(ctx, latestTally, actualBucketData)
+			require.NoError(t, err)
```
Comment (jenlij) [inline pkg/accounting/tally/tally_test.go]: Unless we want to check if SaveBucketTallies worked as expected, we don't need to call it. In TestCalculateAtRestData, we just check the result of CalculateAtRestData, not the result of the db methods.

File: pkg/accounting/tally/tally_test.go (hunk 1 above)
Comment (cam-a) [reply pkg/accounting/tally/tally_test.go]: the size of the metadata is different

File: pkg/accounting/tally/tally_test.go (hunk 2 above)
Comment (cam-a) [reply pkg/accounting/tally/tally_test.go]: The error that this PR fixes was coming from SaveBucketTallies

Comment (cam-a) [reply pkg/accounting/tally/tally_test.go]: unique constraint on attempting to insert into the database

Comment (jenlij) [reply pkg/accounting/tally/tally_test.go]: gotcha

**Pull request merged**